src/frequenz/sdk/actor/power_distributing/result.py

```diff
@@ -0,0 +1,111 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""Results from PowerDistributingActor."""
+
+from abc import ABC
+from typing import Set
+
+from .request import Request
+
+
+class Result(ABC):
+    """Base class for the power distributor result."""
+
+    def __init__(self, request: Request) -> None:
+        """Create class instance.
+
+        Args:
+            request: Initial request
+        """
+        self.request: Request = request
+
+
+class Success(Result):
+    """Send if setting power for all batteries succeed."""
+
+    def __init__(
+        self, request: Request, succeed_power: int, outside_bound: int
+    ) -> None:
+        """Create class instance.
+
+        Args:
+            request: Initial request
+            succeed_power: How much power out of requested were successfully set.
+            outside_bound: How much power was ignored, because was outside power
+                bound. If Request.adjust_power = False, then this value is always 0.
+        """
+        super().__init__(request)
+        self.succeed_power: int = succeed_power
+        self.outside_bound: int = outside_bound
+
+
+class AnyFail(Result):
+    """Send if any request to set power failed by microgrid side."""
+
+    # It is very simple class with only data so it should be ok to disable pylint.
+    # All these results should be dataclass but in python < 3.10 it is risky
+    # to derive after dataclass.
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        request: Request,
+        succeed_power: int,
+        failed_power: int,
+        failed_batteries: Set[int],
+        outside_bound: int,
+    ) -> None:
+        """Create class instance.
+
+        Args:
+            request: Initial_request
+            succeed_power: How much power was successfully set.
+            failed_power: How much power failed.
+            failed_batteries: What batteries failed.
+            outside_bound: How much power was ignored, because was outside power
+                bound. If Request.adjust_power = False, then this value is always 0.
+        """
+        super().__init__(request)
+        self.succeed_power: int = succeed_power
+        self.failed_power: int = failed_power
+        self.failed_batteries: Set[int] = failed_batteries
+        self.outside_bound: int = outside_bound
+
+
+class Error(Result):
+    """Error occurred and power was not set."""
+
+    def __init__(self, request: Request, msg: str) -> None:
+        """Create class instance.
+
+        Args:
+            request: Initial_request
+            msg: Error message telling why error happened.
+        """
+        super().__init__(request)
+        self.msg: str = msg
+
+
+class OutOfBound(Result):
```

Comment:
OutOfBounds?

Reply:
Yes. By default PowerDistributingActor adjust the given power to the bounds.
But Tom told PowerDistributingActor should not adjust power.
I am not agree with it but I add extra field Request.adjust_power: bool. Default to True. If user will set it to False, and requested power will be > bounds, then request won't be executed and OutOfBound will be returned.

Reply:
No,  I meant,  maybe you want to add an `s` at the end,  instead of just `OutOfBound`. :-)

Reply:
No one is using it.
And with async user is not sure when the request will be executd, so it makes no sense.
So maybe I could remove it in some later feature.

Reply:
There is only one bound. 
One for upper bound and one for lower bound :)
In the  message I give only one Bound. So it make sense to make it `OutOfBound`